crates/sui-metric-checker/src/query.rs

```diff
@@ -0,0 +1,88 @@
+// Copyright (c) Mysten Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+use crate::unix_seconds_to_timestamp_string;
+use anyhow::anyhow;
+use base64::{engine::general_purpose, Engine};
+use prometheus_http_query::Client;
+use reqwest::header::{HeaderValue, AUTHORIZATION};
+use tracing::debug;
+
+pub async fn instant_query(
+    auth_header: &str,
+    client: Client,
+    query: &str,
+) -> Result<f64, anyhow::Error> {
+    debug!("Executing {query}");
+    let response = client
+        .query(query)
+        .header(
+            AUTHORIZATION,
+            HeaderValue::from_str(&format!(
+                "Basic {}",
+                general_purpose::STANDARD.encode(auth_header)
+            ))?,
+        )
+        .get()
+        .await?;
```

Comment:
Maybe we could do a few tries in case the query fails? I believe it could be possible having sporadic failures when querying metrics and might be a shame having to start from the beginning.

Reply:
Also can we define some timeout settings for the http client?